render_pdf/src/parser.py

```diff
@@ -0,0 +1,116 @@
+import json
+import sys
+from collections import defaultdict
+from typing import Optional
+from urllib.request import urlopen
+
+from lxml import etree
+from lxml.etree import fromstring
+
+
+def get_body(root):
+    return root.find(tagname("text")).find(tagname("body"))
+
+def get_title(root, alt_root) -> Optional[str]:
+    # God this is ugly # TODO
+    try:
+        return root.find(tagname("head")).text
+    except:
+        try:
+            root = alt_root
+            tag = root.find(tagname("teiHeader"))\
+                    .find(tagname("fileDesc")) \
+                    .find(tagname("titleStmt"))
+            # TODO Check if find('./teiHeader/fileDesc/titleStmt').text syntax will work on tag in lxml.  
+            return tag.find(tagname("title")).text
+        except:
+            # TODO change bare execpt to catch specific errors. 
+            print("[Warning] Couldn't get title.")
+            return 
+
+def get_author(root) -> Optional[str]:
+    try:
+            tag = root.find(tagname("teiHeader"))\
+                    .find(tagname("fileDesc")) \
+                    .find(tagname("titleStmt"))
+            author = tag.find(tagname("author")).text
+    except:
+        return 
+    else:
```

Comment:
`try / except / else` is kind of funky. Can you rewrite this?

something like:
```
author = root.find("./teiHeader/fileDesc/titleStmt/author')
if author:
  return author.text
else:
  return None
```